src/pages/Scan.vue

```diff
@@ -99,20 +100,12 @@ export default {
                 Store.addVoucherCode(this.sponsorCode.toUpperCase()+this.voucherCode,
                     // Success function
                     function(response) {
-                        // Add error message for invalid and fail codes.
-                        var data = response.data;
-
-                        if (data.invalid.length + data.fail.length === 1) {
-                            // single mismatch handler;
-                            if (data.invalid.length > 0) {
-                                this.showFail();
-                                this.setMessage("Please enter a valid voucher code.", constants.MESSAGE_ERROR);
-
-                            } else if (data.fail.length > 0) {
-                                this.showFail();
-                                this.setMessage("It looks like this code has already been added, please double check and try again. If you are still unable to add the voucher code, don\'t worry - you will still receive payment if you send it in with your other vouchers.", constants.MESSAGE_WARNING);
-                            }
-
+                        if (response.error) {
+                            this.showFail();
+                            this.setMessage(response.error, constants.MESSAGE_ERROR);
```

Comment:
Also the messages are part of the "response.data" object not the response itself.